lib/rules/array-indentation.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * @fileoverview Check indentation at the beginning and end of a literal array expression
+ */
+'use strict';
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = function(context) {
+   var sourceCode = context.getSourceCode(),
+       lines = sourceCode.getLines();
+
+   //--------------------------------------------------------------------------
+   // Helpers
+   //--------------------------------------------------------------------------
+
+   function lineIndent(lineNumber) {
+      var line = lines[lineNumber - 1];
+
+      if (line === undefined) {
+         throw new Error('Invalid lineNumber ' + lineNumber + ' supplied to lineIndent');
+      }
+
+      return line[0] === ' ' ? line.match(/[^ \t]/).index : 0;
+   }
+
+   function validateArrayIndentation(node) {
+      var firstLineIndent = lineIndent(node.loc.start.line),
+          lastLineIndent = lineIndent(node.loc.end.line);
+
+      if (firstLineIndent !== lastLineIndent) {
+         context.report({
+            node: node,
+            message: 'Array expressions must begin and end with the same indentation',
+         });
+      }
+   }
+
+   //--------------------------------------------------------------------------
+   // Public
+   //--------------------------------------------------------------------------
+
+   return {
+      'ArrayExpression': function(node) {
+            if (node.loc.start.line !== node.loc.end.line) {
```

Comment:
this whole block is indented wrong

irony:
![](http://static1.lxdcdn.net/images/max/w/806/15b7e89d1cf17d5927267df9856d264f.jpeg)